affinity/cpp-20/d0796r1.md

```diff
@@ -95,395 +96,423 @@ Some systems give additional user control through explicit binding of threads to
 
-The `execution_resource` class is not `CopyConstructible` (C++Std [copyconstructible]).
+### Execution resources
 
-    execution_resource(const execution_resource &) = delete;
-    execution_resource(execution_resource &&) = delete;
-    execution_resource &operator=(const execution_resource &) = delete;
-    execution_resource &operator=(execution_resource &&) = delete;
+An `execution_resource` is a light weight structure which acts as an identifier to particuar piece of hardware within a system. It can be queried for whehter it can allocate memory via `can_place_memory` and whether it can execute work via `can_place_agents`, and for it's name via `name`. An `execution_resource` can also represent other `execution_resource`s, these are refered to as being *members of* that `execution_resource` and can be queried via `resources`. Additionally the `execution_resource` which another is a *member of* can be queried vis `member_of`. An `execution_resource` can also be queried for the concurrency it can provide; the totla number of *threads of execution* supported by that *execution_resource* and all resources it represents.
 
-### `execution_resource` destructor
+> [*Note:* Note that an execution resource is not limited to resources which execute work, but also a general resource where no execution can take place but memory can be allocated such as off-chip memory. *--end note*]
 
-The `execution_resource` class is not `Destructible` (C++Std [destructible]).
+> [*Note:* The intention is that the actual implementation details of a resource topology are described in an execution context when required. This allows the execution resource objects to be lightweight objects that serve as identifiers that are only referenced. *--end note*]
 
-	~execution_resource() = delete;
+### System topology
 
-### `execution_resource` operations
+The system topology is made up of a number of system level `execution_resource`s, which can be queried through `this_system::resource` which returns a `std::vector`. The `execution_resources` available within the system can be initialised dynamically by a runtime library, however must be done so before `main` is called, ass after that point the system topology cannot change.
 
-      size_t concurrency() const noexcept;
+Below is an example of iterating over the system level resources and priniting out it's capabilities.
+
+```cpp
+for (auto res : execution::this_system::resources()) {
+    std::cout << res.name()  `\n`;
+    std::cout << res.can_place_memory() << `\n`;
+    std::cout << res.can_place_agents() << `\n`;
+    std::cout << res.concurrency() << `\n`;
+}
+```
+*Listing 2: Example of querying all the system level execution resources*
```

Comment:
aLL -> all ? not sure weird formatting on my screen

Reply:
I think it's okay, it looks weird on my screen as well.